cloud/azure/datafactory/factory/mode/factoryusage.pm

```diff
@@ -0,0 +1,209 @@
+        resource           => $self->{az_resource},
+        resource_group     => $self->{az_resource_group},
+        resource_namespace => $self->{az_resource_namespace},
+        resource_type      => $self->{az_resource_type},
+        timeframe          => $self->{az_timeframe}
+    );
+
+    $self->{metrics}->{$self->{az_resource}}->{display} = $self->{az_resource};
+    foreach my $aggregation (@{$self->{az_aggregations}}) {
+        foreach my $metric (@{$self->{az_metrics}}) {
+            my $metric_name = lc($metric);
+            $metric_name =~ s/ /_/g;
+            next if (!defined($metric_results{$self->{az_resource}}->{$metric_name}->{lc($aggregation)}) && !defined($aggregation) && !defined($self->{option_results}->{zeroed}));
+            next if (defined($self->{skip_aggregation}->{$metric}->{lc($aggregation)}) && $self->{skip_aggregation}->{$metric}->{lc($aggregation)} == 0);
+            $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}->{display} = lc($aggregation);
+            $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}->{timeframe} = $self->{az_timeframe};
+            $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}->{$metric} =
+                defined($metric_results{$self->{az_resource}}->{$metric_name}->{lc($aggregation)}) ?
+                $metric_results{$self->{az_resource}}->{$metric_name}->{lc($aggregation)} : 0;
+        }
+        # Compute percentages from metrics
+        next if (!defined($self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)}) && !defined($aggregation) && !defined($self->{option_results}->{zeroed}));
+        next if (defined($self->{skip_aggregation}->{statistics}->{lc($aggregation)}) && $self->{skip_aggregation}->{statistics}->{lc($aggregation)} == 0);
+        my $metricsAggregation = $self->{metrics}->{$self->{az_resource}}->{statistics}->{lc($aggregation)};
+        my $factory_size_usage = 0;
+        if (defined($metricsAggregation->{lc('MaxAllowedFactorySizeInGbUnits')})) {
+            my $max_allowed_factory_size_in_gb_units = $metricsAggregation->{lc('MaxAllowedFactorySizeInGbUnits')};
+            if ($max_allowed_factory_size_in_gb_units > 0) {
+                $factory_size_usage = ($metricsAggregation->{lc('FactorySizeInGbUnits')} / $max_allowed_factory_size_in_gb_units) * 100;
+            }
+        }
+        $metricsAggregation->{factory_size_usage} = $factory_size_usage;
+        my $resource_usage = 0;
+        if (defined($metricsAggregation->{lc('MaxAllowedResourceCount')})) {
+            my $max_allowed_resource_count = $metricsAggregation->{lc('MaxAllowedResourceCount')};
+            if ($max_allowed_resource_count > 0) {
+                $resource_usage = ($metricsAggregation->{lc('ResourceCount')} / $max_allowed_resource_count) * 100;
+            }
+        }
+        $metricsAggregation->{resource_usage} = $resource_usage;
+    }
```

Comment:
Let the --zeroed logic possible here. I mean, if no data is returned, the user should know it and choose to set 0 instead of undef explicitely. Think about the cost of the data storage and that the user should definitely choose if he wants to store the 0/fake data.